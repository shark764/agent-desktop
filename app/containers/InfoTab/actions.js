/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * InfoTab actions
 *
 */

import {
  SET_CRM_UNAVAILABLE,
  CLEAR_SEARCH_RESULTS,
  SET_SEARCH_RESULTS,
  CHECK_CONTACT,
  UNCHECK_CONTACT,
  CLEAR_CHECKED_CONTACTS,
  ADD_NOTIFICATION,
  DISMISS_NOTIFICATION,
  SET_LOADING,
  SET_DELETION_PENDING,
  SET_CONFIRMING_DELETE,
  SET_SEARCH_PENDING,
  VALIDATE_CONTACT_LAYOUT_TRANSLATIONS,
} from './constants';

export function setCRMUnavailable(reason) {
  return {
    type: SET_CRM_UNAVAILABLE,
    reason,
  };
}

export function clearSearchResults() {
  return {
    type: CLEAR_SEARCH_RESULTS,
  };
}

export function setSearchResults(response) {
  return {
    type: SET_SEARCH_RESULTS,
    response,
  };
}

export function checkContact(contact) {
  return {
    type: CHECK_CONTACT,
    contact,
  };
}

export function uncheckContact(contact) {
  return {
    type: UNCHECK_CONTACT,
    contact,
  };
}

export function clearCheckedContacts() {
  return {
    type: CLEAR_CHECKED_CONTACTS,
  };
}

export function addNotification(notification) {
  return {
    type: ADD_NOTIFICATION,
    notification,
  };
}

export function dismissNotification(id) {
  return {
    type: DISMISS_NOTIFICATION,
    id,
  };
}

export function setLoading(loading) {
  return {
    type: SET_LOADING,
    loading,
  };
}

export function setDeletionPending(deletionPending) {
  return {
    type: SET_DELETION_PENDING,
    deletionPending,
  };
}

export function setConfirmingDelete(confirmingDelete) {
  return {
    type: SET_CONFIRMING_DELETE,
    confirmingDelete,
  };
}

export function setSearchPending(searchPending) {
  return {
    type: SET_SEARCH_PENDING,
    searchPending,
  };
}

export function validateContactLayoutTranslations() {
  return {
    type: VALIDATE_CONTACT_LAYOUT_TRANSLATIONS,
  };
}
