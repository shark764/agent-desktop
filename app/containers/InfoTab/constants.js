/*
 *
 * InfoTab constants
 *
 */

export const CLEAR_SEARCH_RESULTS = 'app/InfoTab/CLEAR-SEARCH_RESULTS';
export const SET_SEARCH_RESULTS = 'app/InfoTab/SET_SEARCH_RESULTS';
export const CHECK_CONTACT = 'app/InfoTab/CHECK_CONTACT';
export const UNCHECK_CONTACT = 'app/InfoTab/UNCHECK_CONTACT';
export const CLEAR_CHECKED_CONTACTS = 'app/InfoTab/CLEAR_CHECKED_CONTACTS';
export const SET_CONTACT_MODE = 'app/InfoTab/SET_CONTACT_MODE';
export const SET_UNASSIGNED_CONTACT = 'app/InfoTab/SET_UNASSIGNED_CONTACT';
export const ADD_NOTIFICATION = 'app/InfoTab/ADD_NOTIFICATION';
export const DISMISS_NOTIFICATION = 'app/InfoTab/DISMISS_NOTIFICATION';
export const SET_LOADING = 'app/InfoTab/SET_LOADING';
export const SET_DELETION_PENDING = 'app/InfoTab/SET_DELETION_PENDING';
export const SET_CONFIRMING_DELETE = 'app/InfoTab/SET_CONFIRMING_DELETE';
