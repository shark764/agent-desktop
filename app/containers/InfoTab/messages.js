/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 * InfoTab Messages
 *
 * This contains all the text for the InfoTab component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  crmUnavailableTitle: {
    id: 'app.containers.InfoTab.crmUnavailableTitle',
    defaultMessage: 'CRM Unavailable',
  },
  crmUnavailableGeneral: {
    id: 'app.containers.InfoTab.crmUnavailableGeneral',
    defaultMessage: 'The contact management functionality is unavailable.',
  },
  crmUnavailableLayout: {
    id: 'app.containers.InfoTab.crmUnavailableLayout',
    defaultMessage:
      'Failed to load a valid contact layout, check tenant config.',
  },
  crmUnavailableAttribute: {
    id: 'app.containers.InfoTab.crmUnavailableAttribute',
    defaultMessage: 'No contact attributes found, check tenant config.',
  },
  newRecord: {
    id: 'app.containers.InfoTab.newRecord',
    defaultMessage: 'Create New Record',
  },
  loading: {
    id: 'app.containers.InfoTab.loading',
    defaultMessage: 'Loading...',
  },
  created: {
    id: 'app.components.InfoTab.created',
    defaultMessage: 'Contact created.',
  },
  saved: {
    id: 'app.components.InfoTab.saved',
    defaultMessage: 'Changes saved.',
  },
  merged: {
    id: 'app.components.InfoTab.merged',
    defaultMessage: 'Contacts Merged.',
  },
  assigned: {
    id: 'app.components.InfoTab.assigned',
    defaultMessage: 'Interaction assigned to contact.',
  },
  deleted: {
    id: 'app.components.InfoTab.deleted',
    defaultMessage: 'Contact deleted.',
  },
  deletedMultiple: {
    id: 'app.components.InfoTab.deletedMultiple',
    defaultMessage: 'Contacts deleted.',
  },
  notSaved: {
    id: 'app.containers.InfoTab.notSaved',
    defaultMessage: 'Your changes have not been saved.',
  },
  notDeleted: {
    id: 'app.containers.InfoTab.notDeleted',
    defaultMessage: 'Contact deletion failed.',
  },
  notDeletedMultiple: {
    id: 'app.containers.InfoTab.notDeletedMultiple',
    defaultMessage: 'Selected contacts were not deleted.',
  },
  notCreated: {
    id: 'app.containers.InfoTab.notCreated',
    defaultMessage: 'Your new contact has not been created.',
  },
  notAssigned: {
    id: 'app.containers.InfoTab.notAssigned',
    defaultMessage: 'Contact was not assigned to the interaction.',
  },
  notMerged: {
    id: 'app.containers.InfoTab.notMerged',
    defaultMessage: 'Contact merge failed.',
  },
  serverError: {
    id: 'app.containers.InfoTab.serverError',
    defaultMessage: 'Server Error',
  },
  networkError: {
    id: 'app.containers.InfoTab.networkError',
    defaultMessage: 'Network Error',
  },
  contactLayoutMissingLocaleTranslations: {
    id: 'app.containers.InfoTab.contactLayoutMissingLocaleTranslations',
    defaultMessage: 'There are missing translations for the contact layout',
  },
});
