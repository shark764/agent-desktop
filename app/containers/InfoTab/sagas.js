/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { takeEvery, put, select } from 'redux-saga/effects';

import {
  selectLocale,
  selectAttributes,
  selectLayout,
  selectNextNotificationId,
} from './selectors';
import { addNotification } from './actions';
import { VALIDATE_CONTACT_LAYOUT_TRANSLATIONS } from './constants';

export function* goValidateContactLayoutTranslations() {
  const locale = yield select(selectLocale);
  const attributes = yield select(selectAttributes);
  const contactLayout = yield select(selectLayout);
  let isMissingLocaleTranslations;
  contactLayout.layout.forEach((subLayout) => {
    if (subLayout.label[locale] === undefined) {
      isMissingLocaleTranslations = true;
    } else {
      subLayout.attributes.forEach((layoutAttributeId) => {
        const layoutAttribute = attributes.find(
          (attribute) => attribute.id === layoutAttributeId
        );
        if (layoutAttribute && layoutAttribute.label[locale] === undefined) {
          isMissingLocaleTranslations = true;
        }
      });
    }
  });
  if (isMissingLocaleTranslations) {
    const id = yield select(selectNextNotificationId);
    yield put(
      addNotification({
        id,
        messageType: 'contactLayoutMissingLocaleTranslations',
      })
    );
  }
}

export function* validateContactLayoutTranslations() {
  yield takeEvery(
    VALIDATE_CONTACT_LAYOUT_TRANSLATIONS,
    goValidateContactLayoutTranslations
  );
}

export default [validateContactLayoutTranslations];
