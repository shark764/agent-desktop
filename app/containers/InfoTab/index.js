/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * InfoTab
 *
 */

import React from 'react';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import Radium from 'radium';

import BaseComponent from 'components/BaseComponent';
import { setCriticalError } from 'containers/Errors/actions';

import NotificationBanner from 'components/NotificationBanner';
import ContactsControl from 'containers/ContactsControl';
import ContactHeader from 'components/ContactHeader';
import ContactBulkActions from 'components/ContactBulkActions';
import IconSVG from 'components/IconSVG';

import { deleteContacts, setContactAction, addSearchFilter, removeSearchFilter } from 'containers/AgentDesktop/actions';
import { setShowCancelDialog, setShowConfirmDialog, setFormIsDirty, setFormValidity, resetForm } from 'containers/ContactsControl/actions';

import selectInfoTab, { selectCurrentInteraction, selectCheckedContacts,
  selectContactMode, selectEditingContact, selectExpandedQuery, selectNotifications,
  selectNextNotificationId, selectDeletionPending, selectConfirmingDelete, selectCRMUnavailable } from './selectors';
import { clearSearchResults, clearCheckedContacts, setContactMode,
  setEditingContact, addNotification, dismissNotification, setLoading, setDeletionPending, setConfirmingDelete } from './actions';
import messages from './messages';

export class InfoTab extends BaseComponent {

  styles = {
    base: {
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'stretch',
      position: 'relative',
    },
    results: {
      height: '100%',
      alignItems: 'stretch',
    },
    contacts: {
      overflowY: 'auto',
      boxSizing: 'content-box',
      flexGrow: '1',
      flexShrink: '1',
      position: 'relative',
      display: 'flex',
    },
    checkboxSpacing: {
      marginLeft: '-52px',
    },
    loading: {
      display: 'flex',
      justifyContent: 'center',
    },
    loadingIcon: {
      height: '60px',
    },
    notificationBanner: {
      width: 'calc(100% + 76px)',
      position: 'relative',
      left: '-51px',
    },
  };

  componentWillReceiveProps(nextProps) {
    if (!this.state.error) {
      const queryChanged = JSON.stringify(nextProps.selectedInteraction.query) !== JSON.stringify(this.props.selectedInteraction.query);
      const interactionChanged = (nextProps.selectedInteraction.interactionId !== this.props.selectedInteraction.interactionId);
      if (queryChanged || interactionChanged) {
        this.props.clearSearchResults();
        this.props.clearCheckedContacts();
      }
      if (interactionChanged && this.props.contactMode !== 'viewing') {
        this.setNotEditing();
      }
      if (!this.props.deletionPending && nextProps.deletionPending) {
        this.deleteContacts();
      }
    }
  }

  componentWillUnmount() {
    if (!this.state.error) {
      this.props.clearSearchResults();
    }
  }

  componentDidMount() {
    CxEngage.contacts.listAttributes(() => {
      CxEngage.contacts.listLayouts(() => {
        this.props.setLoading(false);
      });
    });
  }

  setDeletionPending = () => {
    this.props.setLoading(true);
    this.props.setDeletionPending(true);
  }

  deleteContacts = () => {
    this.props.deleteContacts(this.props.checkedContacts.map((contact) => contact.id));
    this.props.clearCheckedContacts();
    this.props.setConfirmingDelete(false);
  }

  setSearching = () => {
    this.props.setContactAction(this.props.selectedInteraction.interactionId, 'search');
  }

  newContact = () => {
    this.props.setContactMode('editing');
    this.props.setEditingContact({});
  }

  editContact = () => {
    this.props.setContactMode('editing');
  }

  setViewing = () => {
    this.props.setContactAction(this.props.selectedInteraction.interactionId, 'view');
  }

  clearSearch = () => {
    this.props.removeSearchFilter();
    if (this.props.selectedInteraction.contact !== undefined) {
      this.setViewing();
    }
  }

  setNotEditing = () => {
    this.props.setContactMode('viewing');
    this.props.clearCheckedContacts();
    this.props.setShowCancelDialog(false);
    this.props.setShowConfirmDialog(false);
    this.props.setFormIsDirty(false);
    this.props.setFormValidity(false);
    this.props.resetForm();
  }

  setMerging = () => {
    this.props.setContactMode('merging');
  }

  addNotification = (messageType, isError, errorType) => {
    // TODO: move error logic to Errors reducer/saga and call handleError instead of addNotification if error
    const id = this.props.nextNotificationId;
    if (!isError) {
      setTimeout(() => this.props.dismissNotification(id), 3000);
    }
    this.props.addNotification({ id, errorType, messageType, isError });
  }

  crmUnavailableBanner = () =>
    <div style={[this.props.style, this.styles.base]}>
      <NotificationBanner
        id="crm-unavailable-banner"
        key="crm-unavailable-banner"
        style={this.styles.notificationBanner}
        titleMessage={messages.crmUnavailableTitle}
        descriptionMessage={messages[this.props.crmUnavailable] || messages.crmUnavailableGeneral}
        isError
      />
    </div>

  render() {
    if (this.props.crmUnavailable) {
      return this.crmUnavailableBanner();
    }
    const showBulkActions = this.props.results.length > 0 && this.props.contactMode === 'viewing' && this.props.selectedInteraction.contactAction === 'search';
    const showCheckboxes = this.props.selectedInteraction.contactAction !== 'view' && this.props.contactMode !== 'merging' && this.props.contactMode !== 'editing';
    return (
      <div style={[this.props.style, this.styles.base]}>
        {
          this.props.notifications.length ?
            this.props.notifications.map((notification, index) =>
              <NotificationBanner
                id={`contactNotification${index}`}
                key={notification.id}
                style={this.styles.notificationBanner}
                dismiss={() => this.props.dismissNotification(notification.id)}
                tryAgain={notification.tryAgain}
                titleMessage={messages[notification.errorType]}
                descriptionMessage={messages[notification.messageType]}
                isError={notification.isError}
              />
            )
          : null
        }
        <ContactHeader
          selectedInteraction={this.props.selectedInteraction}
          contactMode={this.props.contactMode}
          editingContactEditing={this.props.editingContact}
          resultsCount={this.props.resultsCount}
          addSearchFilter={this.props.addSearchFilter}
          clearSearch={this.clearSearch}
          query={this.props.expandedQuery}
          removeSearchFilter={this.props.removeSearchFilter}
          editAssignedContact={this.editContact}
          setSearching={this.setSearching}
        />
        {
          this.props.loading ?
            <div id="loadingContainer" style={this.styles.loading}>
              <IconSVG style={this.styles.loadingIcon} id="loadingIcon" name="loading" />
            </div>
          : <div style={[this.styles.contacts, showCheckboxes && this.styles.checkboxSpacing]}>
            <ContactsControl
              newContact={this.newContact}
              results={this.props.results}
              resultsCount={this.props.resultsCount !== undefined ? this.props.resultsCount : -1}
              isCollapsed={this.props.isCollapsed}
              setNotEditing={this.setNotEditing}
              addNotification={this.addNotification}
            />
          </div>
        }
        {
          showBulkActions && !this.props.loading &&
          <ContactBulkActions
            newContact={this.newContact}
            selectedContacts={this.props.checkedContacts}
            deleteContacts={this.setDeletionPending}
            confirmingDelete={this.props.confirmingDelete}
            setMerging={this.setMerging}
            setConfirmingDelete={this.props.setConfirmingDelete}
          />
        }
      </div>
    );
  }
}

InfoTab.propTypes = {
  crmUnavailable: React.PropTypes.string,
  isCollapsed: React.PropTypes.bool.isRequired,
  style: React.PropTypes.object,
  resultsCount: React.PropTypes.number,
  results: React.PropTypes.any,
  addSearchFilter: React.PropTypes.func,
  removeSearchFilter: React.PropTypes.func,
  clearSearchResults: React.PropTypes.func,
  clearCheckedContacts: React.PropTypes.func,
  selectedInteraction: React.PropTypes.object,
  setContactAction: React.PropTypes.func,
  deleteContacts: React.PropTypes.func,
  contactMode: React.PropTypes.string,
  query: React.PropTypes.array,
  deletionPending: React.PropTypes.bool,
  setLoading: React.PropTypes.func,
  setDeletionPending: React.PropTypes.func,
  setConfirmingDelete: React.PropTypes.func,
  setContactMode: React.PropTypes.func,
  setEditingContact: React.PropTypes.func,
  setShowCancelDialog: React.PropTypes.func,
  setFormIsDirty: React.PropTypes.func,
  setFormValidity: React.PropTypes.func,
  resetForm: React.PropTypes.func,
  nextNotificationId: React.PropTypes.number,
  checkedContacts: React.PropTypes.array,
  dismissNotification: React.PropTypes.func,
  addNotification: React.PropTypes.func,
  notifications: React.PropTypes.array,
  editingContact: React.PropTypes.object,
  expandedQuery: React.PropTypes.array,
  loading: React.PropTypes.bool,
  confirmingDelete: React.PropTypes.bool,
};

function mapStateToProps(state, props) {
  return {
    crmUnavailable: selectCRMUnavailable(state, props),
    selectedInteraction: selectCurrentInteraction(state, props),
    checkedContacts: selectCheckedContacts(state, props),
    contactMode: selectContactMode(state, props),
    editingContact: selectEditingContact(state, props),
    expandedQuery: selectExpandedQuery(state, props),
    nextNotificationId: selectNextNotificationId(state, props),
    notifications: selectNotifications(state, props),
    deletionPending: selectDeletionPending(state, props),
    confirmingDelete: selectConfirmingDelete(state, props),
    ...selectInfoTab(state, props),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    setCriticalError: () => dispatch(setCriticalError()),
    clearSearchResults: () => dispatch(clearSearchResults()),
    clearCheckedContacts: () => dispatch(clearCheckedContacts()),
    deleteContacts: (contactIds) => dispatch(deleteContacts(contactIds)),
    setContactMode: (contactMode) => dispatch(setContactMode(contactMode)),
    setEditingContact: (editingContact) => dispatch(setEditingContact(editingContact)),
    addNotification: (notification) => dispatch(addNotification(notification)),
    dismissNotification: (id) => dispatch(dismissNotification(id)),
    setLoading: (loading) => dispatch(setLoading(loading)),
    setDeletionPending: (deletionPending) => dispatch(setDeletionPending(deletionPending)),
    setConfirmingDelete: (confirmingDelete) => dispatch(setConfirmingDelete(confirmingDelete)),
    setShowCancelDialog: (showCancelDialog) => dispatch(setShowCancelDialog(showCancelDialog)),
    setShowConfirmDialog: (showConfirmDialog) => dispatch(setShowConfirmDialog(showConfirmDialog)),
    setContactAction: (interactionId, newAction) => dispatch(setContactAction(interactionId, newAction)),
    setFormIsDirty: (formIsDirty) => dispatch(setFormIsDirty(formIsDirty)),
    setFormValidity: (formIsValid) => dispatch(setFormValidity(formIsValid)),
    addSearchFilter: (filterName, value) => dispatch(addSearchFilter(filterName, value)),
    removeSearchFilter: (filter) => dispatch(removeSearchFilter(filter)),
    resetForm: () => dispatch(resetForm()),
    dispatch,
  };
}

export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(Radium(InfoTab)));
