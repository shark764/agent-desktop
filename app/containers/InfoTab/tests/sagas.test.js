/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { fromJS } from 'immutable';
import { goValidateContactLayoutTranslations } from 'containers/InfoTab/sagas';

const locale = 'en-US';
const contactAttributes = [
  { id: '1', label: { 'en-US': 'Attribute one' } },
  { id: '2', label: { 'en-US': 'Attribute two' } },
];
const contactLayout = {
  layout: [{ label: { 'en-US': 'Contact layout' }, attributes: ['1', '2'] }],
};

let oneTimeNotifications = [];

describe('goValidateContactLayoutTranslations saga', () => {
  let generator;
  describe('if no locales are missing', () => {
    beforeAll(() => {
      generator = goValidateContactLayoutTranslations();
    });
    it('should select locale', () => {
      expect(generator.next()).toMatchSnapshot();
    });
    it('should select attributes', () => {
      expect(generator.next(locale)).toMatchSnapshot();
    });
    it('should select layout', () => {
      expect(generator.next(contactAttributes)).toMatchSnapshot();
    });
    it('should select oneTimeNotifications', () => {
      expect(generator.next(contactLayout)).toMatchSnapshot();
    });
    it('should be finished', () => {
      expect(generator.next(oneTimeNotifications)).toMatchSnapshot();
    });
  });
  describe('if locale is missing in layout', () => {
    let contactLayoutWithMissingTranslation;
    beforeAll(() => {
      contactLayoutWithMissingTranslation = Object.assign({}, contactLayout);
      contactLayoutWithMissingTranslation.layout[0].label = {
        'fr-CA': 'Mise en page des contacts',
      };
      generator = goValidateContactLayoutTranslations();
    });
    it('should select locale', () => {
      expect(generator.next()).toMatchSnapshot();
    });
    it('should select attributes', () => {
      expect(generator.next(locale)).toMatchSnapshot();
    });
    it('should select layout', () => {
      expect(
        generator.next(contactLayoutWithMissingTranslation)
      ).toMatchSnapshot();
    });
    it('should select oneTimeNotifications', () => {
      expect(generator.next(contactLayout)).toMatchSnapshot();
    });
    it('should select the next notification id', () => {
      expect(generator.next(oneTimeNotifications)).toMatchSnapshot();
    });
    it('should add the addOneTimeNotification', () => {
      expect(generator.next()).toMatchSnapshot();
    });
    it('should add the notification', () => {
      expect(generator.next()).toMatchSnapshot();
    });
    it('should be finished', () => {
      expect(generator.next()).toMatchSnapshot();
    });
  });
  describe('if locale is missing in attribute', () => {
    let contactAttributesWithMissingTranslation;
    beforeAll(() => {
      contactAttributesWithMissingTranslation = Object.assign(
        {},
        contactAttributes
      );
      contactAttributesWithMissingTranslation[1].label = {
        'fr-CA': 'Attribut deux',
      };
      generator = goValidateContactLayoutTranslations();
    });
    it('should select locale', () => {
      expect(generator.next()).toMatchSnapshot();
    });
    it('should select attributes', () => {
      expect(generator.next(locale)).toMatchSnapshot();
    });
    it('should select layout', () => {
      expect(
        generator.next(contactAttributesWithMissingTranslation)
      ).toMatchSnapshot();
    });
    it('should select oneTimeNotifications', () => {
      expect(generator.next(contactLayout)).toMatchSnapshot();
    });
    it('should select the next notification id', () => {
      expect(generator.next(oneTimeNotifications)).toMatchSnapshot();
    });
    it('should add the addOneTimeNotification', () => {
      expect(generator.next()).toMatchSnapshot();
    });
    it('should add the notification', () => {
      expect(generator.next()).toMatchSnapshot();
    });
    it('should be finished', () => {
      expect(generator.next()).toMatchSnapshot();
    });
  });
  describe('if locale is missing in attribute and notification has already been shown', () => {
    let contactAttributesWithMissingTranslation;
    beforeAll(() => {
      contactAttributesWithMissingTranslation = Object.assign(
        {},
        contactAttributes
      );
      contactAttributesWithMissingTranslation[1].label = {
        'fr-CA': 'Attribut deux',
      };
      oneTimeNotifications = fromJS([
        { id: '1', label: { 'en-US': 'Attribute one' } },
      ]);
      generator = goValidateContactLayoutTranslations();
    });
    it('should select locale', () => {
      expect(generator.next()).toMatchSnapshot();
    });
    it('should select attributes', () => {
      expect(generator.next(locale)).toMatchSnapshot();
    });
    it('should select layout', () => {
      expect(
        generator.next(contactAttributesWithMissingTranslation)
      ).toMatchSnapshot();
    });
    it('should select oneTimeNotifications', () => {
      expect(generator.next(contactLayout)).toMatchSnapshot();
    });
    it('should be finished', () => {
      expect(generator.next(oneTimeNotifications)).toMatchSnapshot();
    });
  });
});
