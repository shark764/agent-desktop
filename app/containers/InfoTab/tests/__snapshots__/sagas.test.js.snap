// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute and notification has already been shown should be finished 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute and notification has already been shown should select attributes 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute and notification has already been shown should select layout 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute and notification has already been shown should select locale 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute and notification has already been shown should select oneTimeNotifications 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute should add the addOneTimeNotification 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "PUT": Object {
      "action": Object {
        "oneTimeNotification": Object {
          "id": undefined,
          "messageType": "contactLayoutMissingLocaleTranslations",
        },
        "type": "app/InfoTab/ADD_ONE_TIME_NOTIFICATION",
      },
      "channel": null,
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute should add the notification 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "PUT": Object {
      "action": Object {
        "notification": Object {
          "id": undefined,
          "messageType": "contactLayoutMissingLocaleTranslations",
        },
        "type": "app/InfoTab/ADD_NOTIFICATION",
      },
      "channel": null,
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute should be finished 1`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute should select attributes 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute should select layout 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute should select locale 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute should select oneTimeNotifications 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in attribute should select the next notification id 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in layout should add the addOneTimeNotification 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "PUT": Object {
      "action": Object {
        "oneTimeNotification": Object {
          "id": undefined,
          "messageType": "contactLayoutMissingLocaleTranslations",
        },
        "type": "app/InfoTab/ADD_ONE_TIME_NOTIFICATION",
      },
      "channel": null,
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in layout should add the notification 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "PUT": Object {
      "action": Object {
        "notification": Object {
          "id": undefined,
          "messageType": "contactLayoutMissingLocaleTranslations",
        },
        "type": "app/InfoTab/ADD_NOTIFICATION",
      },
      "channel": null,
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in layout should be finished 1`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in layout should select attributes 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in layout should select layout 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in layout should select locale 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in layout should select oneTimeNotifications 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if locale is missing in layout should select the next notification id 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if no locales are missing should be finished 1`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`goValidateContactLayoutTranslations saga if no locales are missing should select attributes 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if no locales are missing should select layout 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if no locales are missing should select locale 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;

exports[`goValidateContactLayoutTranslations saga if no locales are missing should select oneTimeNotifications 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "SELECT": Object {
      "args": Array [],
      "selector": [Function],
    },
  },
}
`;
